server/controllers/auth/login.js

```diff
@@ -0,0 +1,34 @@
+const bcrypt = require('bcrypt');
+const { sign } = require('jsonwebtoken');
+
+const loginValidation = require('../../utils/validation/loginValidation');
+const applicant = require('../../database/models/applicant');
+require('env2')('config.env');
+
+const SECRET = process.env.SECRET_KEY;
+
+const login = (req, res) => {
+  const { email, password } = req.body;
+  loginValidation(email, password)
+    .then((valid) => {
+      if (!valid) throw Error;
+      else {
+        applicant
+          .findOne({ email })
```

Comment:
@ahmadysalah You could try to use `await` instead `then` in this file to remove some of the nesting, which can make the code easier to understand.  Something like this:
```
const { email, password } = req.body;
const valid = await loginValidation( email, password );
if (!valid) ...
else {
  const data = await applicant.findOne({ email });
  bcrypt.compare( .....
}
```

Reply:
i prefer always use await and sync but we are using promises in this project.